contracts/FleetFactoryDeterministic.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity >= 0.5.0 < 0.7.0;
+
+import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+
+contract FleetFactoryDeterministic {
+  GnosisSafeProxyFactory public proxyFactory;
+
+  event FleetDeployed(address indexed owner, address[] fleet);
+
+  constructor(GnosisSafeProxyFactory _proxyFactory) public {
+    proxyFactory = _proxyFactory;
+  }
+
+  function deployFleetWithNonce(address owner, uint256 size, address template, uint256 saltNonce) external {
```

Comment:
Should this function return an array of addresses[] with the all the fleets, in order to get the fleets very easily?